src/stride.js

```diff
@@ -0,0 +1,180 @@
+        if (status === 200) {
+          stream = req.pipe(SubscribeObjectTransform())
+          let streamDestroy = stream.destroy.bind(stream)
+          stream.destroy = () => req.destroy()
+        }
+        resolve({status, stream})
+      })
+    })
+  }
+
+  validateURLForMethod (method, url) {
+    let error = validateURLForMethod(method, url)
+    if (error) throw new Error(`URL ${url} malformed: ${error}`)
+  }
+
+  _callAPIMethod (method, url, args) {
+    this.validateURLForMethod(method, url)
+    let options = this._getRequestOptions(method, url, args)
+    return request(options).then((result) => {
+      let message = result[0]
+      let response = result[1]
+      return {response, status: message.statusCode}
+    })
+  }
+
+  _request (method, url, args) {
+    let options = this._getRequestOptions(method, url, args)
+    return request(options)
+  }
+
+  _getRequestOptions (method, url, body) {
+    return {
+      url,
+      body,
+      method,
+      baseUrl: this._getBaseURL(),
+      json: true,
+      headers: this._getHeaders()
+    }
+  }
+
+  _hyperquest (method, url, cb) {
+    let absoluteURL = this._getBaseURL() + url
+    return hyperquest(absoluteURL, this._getHyperquestOptions(method), cb)
+  }
+
+  _getHyperquestOptions (method) {
+    return {
+      method,
+      headers: this._getHeaders()
+    }
+  }
+
+  _getBaseURL () {
+    return `${this.options.baseURL}/${this.options.version}`
+  }
+
+  _getHeaders () {
+    return {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json',
+      'Authorization': `Basic ${this.base64Token}`
+    }
+  }
+}
+
+function SubscribeObjectTransform() {
+  return through2.obj(function (chunk, encoding, callback) {
+    let objects = chunk.toString('utf8').trim().split('\r\n')
+    for (let obj of objects) {
+      if (obj = obj.trim()) {
+        obj = JSON.parse(obj)
+        if (obj && typeof obj === 'object') this.push(obj)
+      }
+    }
+    callback()
+  })
+}
+
+function validateURLForMethod (method, url) {
+  let error = validateURL(url)
+  if (error) return error
+
+  let supported = false
+  let patterns = SupportedURLPatterns[method]
+  for (let pattern of patterns) {
+    if (pattern.test(url)) supported = true
+  }
+  if (!supported) return `URL not supported for the ${method} method`
+
+  return null
+}
+
+function validateURL (url) {
+  for (var key in ValidationRegExps) {
+    if (!ValidationRegExps[key].test(url)) return ValidationMessages[key]
+  }
+  return null
+}
+
+const ValidationRegExps = {
+  hasLeadingSlash: /^\//,
+  missingVersionNumber: /^\/[^v]/,
+  isValidEndpoint: /^(\/collect|\/process|\/analyze)/,
+}
+
+const ValidationMessages = {
+  hasLeadingSlash: 'Endpoints should have leading slash. e.g. `/collect`',
+  missingVersionNumber: 'Please omit the version number. e.g. `/collect`',
+  isValidEndpoint: 'Only the `/collect`, `/process`, and `/analyze` endpoints are supported',
+}
+
+const SupportedURLPatterns = {
```

Comment:
@usmanm, think this is too aggressive for validation? Basically it whitelists all the available endpoints.